test/Integration/test-csharp/GetProductColumnTypesSerializationAsyncEnumerable.cs

```diff
@@ -13,30 +12,29 @@
 
 namespace Microsoft.Azure.WebJobs.Extensions.Sql.Samples.InputBindingSamples
 {
-    public static class GetProductsColumnTypesSerializationDifferentCulture
+    public static class GetProductsColumnTypesSerializationAsyncEnumerable
     {
         /// <summary>
         /// This function verifies that serializing an item with various data types
-        /// works when the language is set to a non-enUS language.
-        /// Note this uses IAsyncEnumerable because IEnumerable serializes the entire table directly,
-        /// instead of each item one by one (which is where issues can occur)
+        /// works when using IAsyncEnumerable.
         /// </summary>
-        [FunctionName(nameof(GetProductsColumnTypesSerializationDifferentCulture))]
+        [FunctionName(nameof(GetProductsColumnTypesSerializationAsyncEnumerable))]
         public static async Task<IActionResult> Run(
-            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "getproducts-columntypesserializationdifferentculture")]
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "getproducts-columntypesserializationasyncenumerable")]
             HttpRequest req,
             [Sql("SELECT * FROM [dbo].[ProductsColumnTypes]",
                 CommandType = System.Data.CommandType.Text,
                 ConnectionStringSetting = "SqlConnectionString")]
             IAsyncEnumerable<ProductColumnTypes> products,
             ILogger log)
         {
-            CultureInfo.CurrentCulture = new CultureInfo("it-IT", false);
```

Comment:
@Charles-Gagnon When I checked the logs, the CurrentCulture didn't affect how the datetime was serialized (I believe it only affects strings). I removed the CurrentCulture and am using this sample to test IAsyncEnumerable and the GetProductsColumnTypes sample to test IEnumerable. Please let me know if there is something I am missing here.

Reply:
So you're correct in that it doesn't affect how it's serialized - the issue that we ran into was that we were calling ToString on the Date objects in our extension code - which does adhere to the CurrentCulture. Then when that was converted back into a DateTime for the POCO object it failed at that point with this error 

`[2022-08-25T07:31:54.100Z] System.Private.CoreLib: Exception while executing function: Function1. Newtonsoft.Json: Could not convert string to DateTime: 25/08/2022 07:19:20. Path 'LastExecutionDate', line 1, position 51.`

The issue was fixed so removing the CurrentCulture being set here isn't going to change anything (since I removed the ToString completely) - but it means that if we were to revert that fix we wouldn't catch it.

Basically this test is just verifying that changing the culture of the function doesn't break any of the serialization/deserialization stuff. 

What I'd suggest is having the function take in an HTTP parameter that tells it what language to use (with no value no changing anything). Then you should make multiple versions of the test using this that specify other languages to use (for now just one we know fails like it-IT is fine in addition to default en-us)

Reply:
Added back the culture changes - thanks for the explanation!